### ISSUE TITLE
Interest in custom Branch drawing symbols 🤔

### ISSUE THREAD

**Issue by bew** [wezterm/wezterm#6328]
Hello Wez o/

I found out about [this PR](https://github.com/kovidgoyal/kitty/pull/7681) (and [its followup](https://github.com/kovidgoyal/kitty/pull/7805)) to kitty terminal to add custom branch-drawing symbols, and was wondering if you'd accept a PR to add similar symbols to Wezterm 🤔

![](https://github.com/user-attachments/assets/4eb16021-7169-4ada-99e5-672a062e52f2)

(repeating part of that PR here…) These symbols are on an arbitrary range (0xf5d0 - 0xf5fb) in a Unicode Private Use Area that is a reasonable distance from the previous icon set in Nerd icons.

cc @rbong
Would you say the set of chars is final and ready to be impl in other terminals? Or are there more to be added soon™?
I see some discussion about these symbols [here](https://github.com/SuperBo/fugit2.nvim/discussions/88), are there other?

---
related: _I'm thinking about first implementing the custom chars 'selector' we talked about [a long time ago in 2022…](https://github.com/wez/wezterm/issues/1957#issuecomment-1120399027), and these new chars could then be easily disabled if needed_ 🤔

**Comment by rbong** [comment]
>Would you say the set of chars is final and ready to be impl in other terminals? Or are there more to be added soon™?

You can draw any kind of directed acyclic graph in any orientation using these characters, including in orientations not typically used by Git branch viewers, and with connections working in a couple different ways... so I would say they're ready and future proof.

However one type of symbol I have not added is branches "jumping over" other branches. This could be up to 4 new symbols. These are not a necessity - you can always just draw "disconnected" branches, which is what every terminal branch viewer does right now and is much less visually cluttered. I have no explicit plans to add these symbols - however since some graphical branch viewers use these types of symbols, they're in the back of my mind and I wonder if they're needed.

There has also been discussion about different types of commit symbols. I am not convinced we need any more and have no plans to add more myself... The only proposed symbols right now are not typically required in a graph view, just a linear view, so I remain unconvinced that they're useful. I believe there will always be ideas for more commit symbols, and increasing the total number of symbols we have now by nearly 50% of the current total is not worth it. What we have is sufficient.

Feedback very much welcome.

**Comment by wez** [comment]
Yeah, I'll take a PR for this! While you're in this mode, the spinner glyphs from https://github.com/ryanoasis/nerd-fonts/issues/1345 are appealing to me in order to support #6581 
I think those have been merged into nerdfonts but not released, and I'd like to use them for indeterminate progress state :)

**Comment by wez** [comment]
Thanks to @BenBergman, all the new glyphs discussed here are now present in `main`!
